selfservice/flow/login/hook.go

```diff
@@ -257,6 +257,16 @@ func (e *HookExecutor) PostLoginHook(
 		// Browser flows rely on cookies. Adding tokens in the mix will confuse consumers.
 		s.Token = ""
 
+		// If we detect that whoami would require a higher AAL, we redirect!
+		if _, err := e.requiresAAL2(r, s, a); err != nil {
+			if aalErr := new(session.ErrAALNotSatisfied); errors.As(err, &aalErr) {
+				span.SetAttributes(attribute.String("return_to", aalErr.RedirectTo), attribute.String("redirect_reason", "requires aal2"))
+				e.d.Writer().WriteError(w, r, flow.NewBrowserLocationChangeRequiredError(aalErr.RedirectTo))
```

Comment:
This would loose the OAuth2 URL though, would it not? So the user would end up in the "Please enter your second factor" but after submitting they would end up at the default return URL. Instead however, they should complete the OAuth2 flow.

Reply:
Nope, user ends up at correct endpoint because of login_challenge also getting passed down - this uses the same, working(?), logic as the form body response, and after testing it works fine...

Reply:
I see, that's great! Would it be possible for you to add a test case to make sure this doesn't break in the future? :)

Reply:
Made my tests more verbose to check for login_challenge anyway